File: 170-two-sum-iii-data-structure-design/170-two-sum-iii-data-structure-design.java
class TwoSum {
    Map<Integer, Integer> map;
    public TwoSum() {
        map = new HashMap<>();
    }
    
    public void add(int number) {
        map.put(number, map.getOrDefault(number, 0) + 1);
    }
    
    public boolean find(int value) {
        for(int k : map.keySet()) {
            if(map.containsKey(value - k) ) {
                if(k != value - k) return true;
                else if(k == value- k) {
                    if(map.get(k) > 1) return true;
                }
            }
        }
        return false;
    }
}

/**
 * Your TwoSum object will be instantiated and called as such:
 * TwoSum obj = new TwoSum();
 * obj.add(number);
 * boolean param_2 = obj.find(value);
 */

File: 170-two-sum-iii-data-structure-design/170-two-sum-iii-data-structure-design.py
class TwoSum:

    def __init__(self):
        self.dic= defaultdict(int)

    def add(self, number: int) -> None:
        self.dic[number] += 1

    def find(self, value: int) -> bool:
        for num in self.dic.keys():
            comple = value - num
            if comple != num:
                if comple in self.dic:
                    return True
            elif self.dic[num] > 1:
                return True
        return False
            


# Your TwoSum object will be instantiated and called as such:
# obj = TwoSum()
# obj.add(number)
# param_2 = obj.find(value)